cmake/fw_zip.cmake

```diff
@@ -30,7 +30,7 @@ function(generate_dfu_zip)
     "type=${GENZIP_TYPE}"
     "board=${CONFIG_BOARD}"
     "soc=${CONFIG_SOC}"
-    DEPENDS ${GENZIP_TARGET}
+    DEPENDS ${GENZIP_TARGET} ${GENZIP_BIN_FILES}
```

Comment:
The `generate_dfu_zip()` is called in other places like here:
https://github.com/nrfconnect/sdk-nrf/blob/5812695752e30ecb53a09b9953c98acd74c32cf3/modules/mcuboot/CMakeLists.txt#L228-L229

in which cases the `GENZIP_TARGET` is mcuboot_sign_target, which is defined here:
https://github.com/nrfconnect/sdk-nrf/blob/5812695752e30ecb53a09b9953c98acd74c32cf3/modules/mcuboot/CMakeLists.txt#L217

mcuboot_sign_target is dependent on `app_signed_hex` variable content provided with `sign()`, but the DEPENDS to sign only seems to be given hex files https://github.com/nrfconnect/sdk-nrf/blob/5812695752e30ecb53a09b9953c98acd74c32cf3/modules/mcuboot/CMakeLists.txt#L125-L126

I would assume that a rebuild results in new hexfiles being created, and thus the `mcuboot_sign_target` should be invoked.

Also, a target with name deduced from `GENZIP_OUTPUT` is created, and that is marked: `ALL`.
https://github.com/nrfconnect/sdk-nrf/blob/5812695752e30ecb53a09b9953c98acd74c32cf3/cmake/fw_zip.cmake#L39-L43

Have you investigated why things are not being brought up-to-date, or have you just discovered that adding the `${GENZIP_BIN_FILES}` as dependency works ?

I'm a little concerned that there could be other files that are not brought up-to-date that we simply have not discovered yet.

Reply:
I believe the fix to be valid.

It is the same as

https://github.com/zephyrproject-rtos/zephyr/commit/5962aab42f4dca014acf6013ef655c98295bbf89

Reply:
> I believe the fix to be valid.

Except for the fact that in this case there are apparently provided files that may not be built in the current setup.
Which means that CI is failing of you notice.

Reply:
with current setup I mean the (current) app CMake instance.
Additional bin files are produced by child images, and thus unknown in current.

And we don't know prior to invoking the child image, which files it will produce.

Reply:
a small update.

All bin files are created in the same CMake instance that creates the zip, so all knowledge is know, we just need to make it available.

What is created in the child builds are the hex files, such as `PM_SIGNED_APP_HEX` (`signed_by_b0_app.hex`).
But that is  converted here https://github.com/nrfconnect/sdk-nrf/blob/74938a54b79010f8d146655f011e4af4042ea240/modules/mcuboot/CMakeLists.txt#L57

What was missing was the output from the validation_data.py custom command.